### ISSUE TITLE
Completion popup get filled with random stuff

### ISSUE THREAD

**Issue by RUSshy** [d-language-server/vscode-dlang#14]
Sometimes, i don't know why, but completion popup is filled with random stuff

![image](https://user-images.githubusercontent.com/18348637/59129519-5ad72300-896d-11e9-807f-4b923c81f6b9.png)

``ta`` should only have this:

![image](https://user-images.githubusercontent.com/18348637/59129540-6c202f80-896d-11e9-9c5b-bf795d518bfb.png)

**Comment by RUSshy** [comment]
Hmm looks like it doesn't understand generics:

![image](https://user-images.githubusercontent.com/18348637/59139781-c7feae80-8995-11e9-9db6-c1203378af9b.png)

**Comment by RUSshy** [comment]
I think this issue should be moved to DLS repository

**Comment by RUSshy** [comment]
Looks like the issue is with ``auto`` it can't deduce the type:

![image](https://user-images.githubusercontent.com/18348637/59229219-1d2b0200-8bda-11e9-922b-d7789295a786.png)

**Comment by LaurentTreguier** [comment]
Yes, DCD sometimes has trouble with inferred types. I don't know if this can be improved (well, technically it can be I think, but I'm not sure how to go around it).
To be honest, I'm really hoping that the Google Summer Of Code project about creating a language server based on DMD ends up replacing DLS and Serve-d. Things like type resolution shouldn't be a problem when using the DMD frontend to parse code.
